pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -442,10 +739,102 @@ def timestamp(self) -> datetime:
         return self._timestamp
 
 
-def _map_usage(response: chat.ChatCompletion | ChatCompletionChunk) -> usage.Usage:
+@dataclass
+class OpenAIResponsesStreamedResponse(StreamedResponse):
+    """Implementation of `StreamedResponse` for OpenAI Responses API."""
+
+    _model_name: OpenAIModelName
+    _response: AsyncIterable[responses.ResponseStreamEvent]
+    _timestamp: datetime
+
+    async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
+        async for chunk in self._response:
+            if isinstance(chunk, responses.ResponseCompletedEvent):
+                self._usage += _map_usage(chunk.response)
+
+            elif isinstance(chunk, responses.ResponseContentPartAddedEvent):
+                pass  # there's nothing we need to do here
+
+            elif isinstance(chunk, responses.ResponseContentPartDoneEvent):
+                pass  # there's nothing we need to do here
+
+            elif isinstance(chunk, responses.ResponseCreatedEvent):
+                pass  # there's nothing we need to do here
```

Comment:
What are these parts? Are we sure we shouldn't use them, or if we don't know how to use them, then error.

Could you add a comment explaining what each part is to make it easy to understand in future.

Reply:
I specifically took @dmontagu time yesterday on this part of the code. We went through the events together.

Reply:
please can you add a TODO to review and either warn or error if we get parts we don't yet support.

Reply:
> please can you add a TODO to review and either warn or error if we get parts we don't yet support.

We already have a warning below on events that we don't do anything. Those events here are the ones we checked and we are not supposed to do anything. At least, there's no need given the other events we already handle.